File: Greedy/942. DI String Match.cpp
class Solution {
public:
    vector<int> diStringMatch(string s) {
        int n = s.size();
        vector<int> ans(n + 1, 0);
        
        int c = 0;
        for (int i = 0; i < n; i++){
            if (s[i] == 'I'){
                ans[i] = c;
                c++;
            }
        }
        
        ans[n] = c;
        c++;
        
        for (int i = n - 1; i > -1; i--){
            if (s[i] == 'D'){
                ans[i] = c;
                c++;
            }
        }
        
        return ans;
    }
};

File: Greedy/942. DI String Match.py
class Solution:
    def diStringMatch(self, s: str) -> List[int]:
        n = len(s)
        
        ans = [0] * (n + 1)
        c = 0
        
        for i in range(n):
            if s[i] == 'I':
                ans[i] = c
                c += 1
        ans[n] = c
        c += 1
        
        for i in range(n - 1, -1, -1):
            if s[i] == 'D':
                ans[i] = c
                c += 1
        return ans